server/attributes.c

```diff
@@ -110,6 +126,13 @@ static struct attr *create_attribute(
         .mutex = PTHREAD_MUTEX_INITIALIZER,
         .update_index = malloc(count * sizeof(uint64_t)),
     };
+    if (methods->in_change_set && methods->polled_change_set) {
+        attr->last_values = malloc(count * sizeof(char *));
+        for (unsigned int i = 0; i < count; i ++) {
+            attr->last_values[i] = malloc(count * MAX_RESULT_LENGTH);
```

Comment:
Am pretty certain you're double allocating here, it should be enough to allocate:
```c
attr->last_values[i] = malloc(MAX_RESULT_LENGTH);
```

Reply:
Thanks for spotting that, it is fixed